src/Microsoft.Azure.WebJobs.Host/Executors/TaskMethodInvoker.cs

```diff
@@ -18,55 +17,7 @@ public TaskMethodInvoker(Func<TReflected, object[], Task<TReturnType>> lambda)
 
         public Task<TReturnType> InvokeAsync(TReflected instance, object[] arguments)
         {
-            Task<TReturnType> task = _lambda.Invoke(instance, arguments);
-            ThrowIfWrappedTaskInstance(task);
-            return task;
-        }
-
-        private static void ThrowIfWrappedTaskInstance(Task task)
```

Comment:
If someone were to return a nested task, it just won't work like they expect. I don't think checking for this on every invocation is warranted.